src/xwayland/XWayland.cpp

```diff
@@ -1,21 +1,36 @@
 #include "XWayland.hpp"
+#include "../Compositor.hpp"
 #include "../debug/Log.hpp"
+#include "../helpers/fs/FsUtils.hpp"
 
-CXWayland::CXWayland(const bool enabled) {
+CXWayland::CXWayland(const bool wants_enabled) {
 #ifndef NO_XWAYLAND
-    Debug::log(LOG, "Starting up the XWayland server");
-
-    pServer = std::make_unique<CXWaylandServer>();
-
-    if (!enabled) {
+    // Disable Xwayland and clean up if the user disabled it.
+    if (!wants_enabled) {
+        Debug::log(LOG, "XWayland has been disabled, cleaning up...");
+        for (auto& w : g_pCompositor->m_vWindows) {
+            if (w->m_pXDGSurface || !w->m_bIsX11)
+                continue;
+            g_pCompositor->closeWindow(w);
+        }
         unsetenv("DISPLAY");
+        m_benabled = false;
+        return;
+    }
+    if (!NFsUtils::executableExistsInPath("Xwayland")) {
```

Comment:
can you put a newline above this if? Having } and if without a break is a bit cluttery